src/Propel/Common/Config/Loader/FileLoader.php

```diff
@@ -61,6 +61,8 @@ public function __construct(FileLocatorInterface $locator = null)
      * Replaces parameter placeholders (%name%) by their values for all parameters.
      *
      * @param array $configuration The configuration array to resolve
+     *
+     * @return array|void
```

Comment:
As already said, I disagree on your interpretation on that. Let's move on with the actual purpose of this PR.